File: 프로그래머스/1/340199. ［PCCE 기출문제］ 9번 ／ 지폐 접기/［PCCE 기출문제］ 9번 ／ 지폐 접기.java
class Solution {
    public int solution(int[] wallet, int[] bill) {
        int answer = 0;
        while (compare(wallet, bill)){
            fold(bill);
            answer++;
        }
        return answer;
    }
    private static void fold(int[] bill){
        if(bill[0] >= bill[1]){
            bill[0] /= 2;
        } else {
            bill[1] /= 2;
        }
    }
    private static boolean compare(int[] wallet, int[] bill){
        if(Math.max(wallet[0], wallet[1]) < Math.max(bill[0], bill[1])){
            return true;
        }
           if(Math.min(wallet[0], wallet[1]) < Math.min(bill[0], bill[1])){
               return true;
           }
           return false;
    }
}
           

File: 프로그래머스/1/340199. ［PCCE 기출문제］ 9번 ／ 지폐 접기/［PCCE 기출문제］ 9번 ／ 지폐 접기.py
def solution(wallet, bill):
    
    def fold(bill):
        if bill[0] >= bill[1]:
            bill[0] //= 2
        else:
            bill[1] //= 2
    
    def compare(wallet, bill):
        if max(wallet[0], wallet[1]) < max(bill[0], bill[1]):
            return True
        if min(wallet[0], wallet[1]) < min(bill[0], bill[1]):
            return True
        return False
    
    answer = 0
    while compare(wallet, bill):
        fold(bill)
        answer += 1
    return answer